source/gui/addonStoreGui/viewModels/store.py

```diff
@@ -612,6 +619,42 @@ def _getAvailableAddonsInBG(self):
 		core.callLater(delay=0, callable=self.detailsVM.updated.notify, addonDetailsVM=self.detailsVM)
 		log.debug("completed refresh")
 
+	def cancelDownloadAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+		try:
+			addonDataManager._downloadsPendingCompletion.remove(listItemVM)
+		except KeyError:
+			log.debug("Download already completed")

```

Comment:
if this is the case, shouldn't this call `cancelAddonInstall` or similar to ensure the install is cancelled correctly?

Reply:
I'd encourage splitting out the logic for downloaded, pending install add-ons from `cancelAddonInstall` to handle this better

Reply:
please ensure this calls `_cancelPendingInstallForAddon` if the download has completed

Reply:
```suggestion
			log.debug("Download already completed")
			return self._cancelPendingInstallForAddon(listItemVM)
```